astropy/wcs/tests/test_wcs.py

```diff
@@ -1934,3 +1943,412 @@ def test_DistortionLookupTable():
+            assert_allclose(wcs_prog.all_pix2world(1, 1, 1), [10, 20])
+        elif function == "all_world2pix":
+            assert_allclose(wcs_prog.all_world2pix(10, 20, 1), [1, 1])
+
+    def test_multiple_set(self):
+        # Make sure that things work fine if we make the WCS programmatically
+        # and not from a header
+
+        wcs_prog = wcs.WCS(naxis=2, preserve_units=True)
+        wcs_prog.wcs.ctype = "RA---TAN", "DEC--TAN"
+        wcs_prog.wcs.cunit = "arcsec", "arcsec"
+        wcs_prog.wcs.crval = 10, 20
+        wcs_prog.wcs.cdelt = 1, 2
+        wcs_prog.wcs.crpix = 1, 1
+
+        assert list(wcs_prog.wcs.cunit) == ["arcsec", "arcsec"]
+        assert_allclose(wcs_prog.wcs.crval, [10, 20])
+        assert_allclose(wcs_prog.wcs.cdelt, [1, 2])
+
+        wcs_prog.wcs.set()
+
+        with pytest.raises(
+            AttributeError,
+            match="Original units have already been set, cannot change them",
+        ):
+            wcs_prog.wcs.cunit = "arcmin", "arcmin"
+
+        assert list(wcs_prog.wcs.cunit) == ["arcsec", "arcsec"]
+        assert_allclose(wcs_prog.wcs.crval, [10, 20])
+        assert_allclose(wcs_prog.wcs.cdelt, [1, 2])
+
+        wcs_prog.wcs.set()
+
+        assert list(wcs_prog.wcs.cunit) == ["arcsec", "arcsec"]
+        assert_allclose(wcs_prog.wcs.crval, [10, 20])
+        assert_allclose(wcs_prog.wcs.cdelt, [1, 2])
+
+    def test_str_and_repr(self):
+        expected = (
+            "WCS Keywords\n"
+            "\n"
+            "Number of WCS axes: 5\n"
+            "CTYPE : 'RA---TAN' 'FREQ' 'DEC--TAN' 'WAVE' 'OFFSET'\n"
+            "CUNIT : arcsec GHz arcsec nm arcmin\n"
+            "CRVAL : 4.0 5.0 6.0 7.0 8.0\n"
+            "CRPIX : 1.0 2.0 3.0 4.0 5.0\n"
+            "PC1_1 PC1_2 PC1_3 PC1_4 PC1_5  : 1.0 0.0 0.0 0.0 0.0\n"
+            "PC2_1 PC2_2 PC2_3 PC2_4 PC2_5  : 0.0 1.0 0.0 0.0 0.0\n"
+            "PC3_1 PC3_2 PC3_3 PC3_4 PC3_5  : 0.0 0.0 1.0 0.0 0.0\n"
+            "PC4_1 PC4_2 PC4_3 PC4_4 PC4_5  : 0.0 0.0 0.0 1.0 0.0\n"
+            "PC5_1 PC5_2 PC5_3 PC5_4 PC5_5  : 0.0 0.0 0.0 0.0 1.0\n"
+            "CDELT : 4.0 3.0 2.0 1.0 6.0\n"
+            "NAXIS : 0  0  0  0  0"
+        )
+        assert str(self.wcs_preserve) == expected
+        assert repr(self.wcs_preserve) == expected
+
+    def test_print_contents(self, capfd):
+        self.wcs_preserve.wcs.print_contents()
+        captured = capfd.readouterr()
+
+        expected = """
+      cdelt: 0xADDR
+               4.0000       3.0000       2.0000       1.0000       6.0000
+      crval: 0xADDR
+               4.0000       5.0000       6.0000       7.0000       8.0000
+      cunit: 0xADDR
+             "arcsec"
+             "GHz"
+             "arcsec"
+             "nm"
+             "arcmin"
+             """.strip()
+
+        assert expected in "\n".join(
+            [x.rstrip() for x in strip_memory_addresses(captured.out).splitlines()]
+        )
+
+    def test_tab(self):
```

Comment:
Please change the test name to something more detailed: `test_wcstab_cunit_conversion` or similar. Currently it is too generic, i.e., suggesting it is testing that support `-TAB` WCS in general works.

Reply:
Just to be clear, the full name of the test is ``TestPreserveUnits.test_tab``. However, I have renamed it anyway following your suggestion.